### PR TITLE
Extracting out tokenization alignment functions

### PR DESCRIPTION
**Pull request by pruksmhc**
Changes:
Extracting out tokenization alignment functions from probing/ to src/utils/retokenize.py. This will allow non-probing tasks to also use these tokenization alignment methods.

### CODE REVIEW

File: src/utils/retokenize.py [hunk 1]
```diff
@@ -207,3 +210,61 @@ def project_span(self, start, end) -> Tuple[int, int]:
...
+    bpe_tokens = openai_utils.tokenize(text)
+    ta = TokenAligner(eow_tokens, bpe_tokens)
+    return ta, bpe_tokens
+
+def align_bert(text: Text, model_name: str) -> Tuple[TokenAligner, List[Text]]:
+    # If using lowercase, do this for the source tokens for better matching.
+    do_lower_case = model_name.endswith('uncased')
+    bow_tokens = space_tokenize_with_bow(text.lower() if do_lower_case else
+                                         text)
+    bert_tokenizer = get_tokenizer(bow_tokens)
```
Comment (iftenney) [inline src/utils/retokenize.py]: shouldn't this be `get_tokenizer(model_name)` ?

**Comment by iftenney** [comment]
Not sure if this is ready to review yet, but can you also refactor `retokenize_edge_data.py` to import these functions so they aren't defined twice?

Also, can you test with some sample data for each tokenizer (SPR2 should be small and run quick) and compare to the old version? Thanks!

File: src/utils/retokenize.py [hunk 2]
```diff
@@ -207,3 +210,61 @@ def project_span(self, start, end) -> Tuple[int, int]:
...
+def align_moses(text: Text) -> Tuple[TokenAligner, List[Text]]:
+    MosesTokenizer = get_tokenizer("MosesTokenizer")
+    moses_tokens = MosesTokenizer.tokenize(text)
+    cleaned_moses_tokens = utils.unescape_moses(moses_tokens)
+    ta = TokenAligner(text, cleaned_moses_tokens)
+    return ta, moses_tokens
+
+def align_openai(text: Text) -> Tuple[TokenAligner, List[Text]]:
+    eow_tokens = space_tokenize_with_eow(text)
+    openai_utils = get_tokenizer("OpenAI.bpe_tokens")
```
Comment (iftenney) [inline src/utils/retokenize.py]: `get_tokenizer("OpenAI.BPE")` ?

**Comment by pruksmhc** [comment]
Tested on spr2!

**Pull request merged**